<?xml version="1.0"?>
<launch>
    <arg name="gpuid" default="0" />
    <arg name="front_camera" default="realsense_front" />
    <arg name="color_image" default="/$(arg front_camera)/color/image_rect_color/compressed" />
    <arg name="depth_image" default="/$(arg front_camera)/depth/image_rect_raw/compressed" />
    <arg name="model" default="$(find tfpose_ros)/models/graph/mobilenet_thin/graph_opt.pb" />
    <arg name="resolution" default="432x368" />
    <arg name="visualize" default="true" />
    <arg name="hertz" default="30" />

    <node name="pose_estimator" pkg="tfpose_ros" type="broadcaster_ros.py" output="screen" required="true">
        <env name="CUDA_VISIBLE_DEVICES" value="$(arg gpuid)" />
        <param name="model" value="$(arg model)" />
        <param name="resolution" value="$(arg resolution)" />
        <param name="hertz" value="$(arg hertz)"/>
        <remap from="~color" to="$(arg color_image)" />
        <remap from="~depth" to="$(arg depth_image)" />
    </node>

    <node if="$(arg visualize)" name="pose_estimator_vis" pkg="tfpose_ros" type="visualization.py" output="screen" required="true">
        <param name="camera" value="$(arg color_image)" />
        <param name="pose" value="/pose_estimator/pose" />
    </node>

</launch>
